internal/services/ipfs/ipfs.go

```diff
@@ -0,0 +1,79 @@
+package ipfs
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/DIMO-Network/devices-api/internal/config"
+)
+
+type IPFS struct {
+	url    string
+	client *http.Client
+}
+
+type ipfsResponse struct {
+	Success bool   `json:"success"`
+	CID     string `json:"cid"`
+}
+
+func NewIPFSLoader(settings *config.Settings) *IPFS {
+	return &IPFS{
+		client: &http.Client{
+			Timeout: 5 * time.Second,
+		},
+		url: settings.IPFSURL,
+	}
+}
+
+func (i *IPFS) UploadImage(img string) (string, error) {
+	var respb ipfsResponse
+	imageData := strings.TrimPrefix(img, "data:image/png;base64,")
+	image, err := base64.StdEncoding.DecodeString(imageData)
+	if err != nil {
+		return respb.CID, err
+	}
+
+	if len(image) == 0 {
+		return respb.CID, fmt.Errorf("Empty image field.")
```

Comment:
Piling onto the linter: we can do `errors.New` if there's nothing to interpolate.